### PR TITLE
Include Priority in Camunda service filters

### LINKED ISSUE #20495: Include Priority Field in Tasklist V2 C8 Unified API
In the new C8 Unified API, UserTasks will be directly filtered from Zeebe. Include the new Priority field in the filters exposed by Zeebe in _UserTaskFilterTransformer_.

### Acceptance Criteria

UserTasks can be queried via the V2 API filtered by _Priority_.

### PR DESCRIPTION
**Pull request by panagiotisgts**
## Description

Include the new _Priority_ field in the C8 Unified API search filters & sorting parameters for UserTasks.

## Related issues

closes #20495

### CODE REVIEW

File: service/src/main/java/io/camunda/service/entities/UserTaskEntity.java
Comment (nathansandi) [inline service/src/main/java/io/camunda/service/entities/UserTaskEntity.java:35]: ❓ The range of priority is [0,100]? 
❓ If in case yes, it can be Integer?

File: service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java
Comment (nathansandi) [inline service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java:39]: 💭  Reminding me to create a follow up issue and add the priority fields for Client/Rest API 

As this is on the review, wont add it now to avoid large scope

File: service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java
Comment (nathansandi) [inline service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:55]: ❓ Is this field already included on "tasklist-task" indice?

File: service/src/test/java/io/camunda/service/query/filter/UserTaskFilterTest.java
Comment (nathansandi) [inline service/src/test/java/io/camunda/service/query/filter/UserTaskFilterTest.java:478]: 🔧  Is it possible add the test cases other operations for gt,... ? 

In order to check if the filter is applied for other cases

Or the scope is only equals from now?

File: service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java
Comment (panagiotisgts) [reply service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:55]: Will be when the time to merge this PR comes

File: service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java
Comment (panagiotisgts) [reply service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java:39]: It's part of https://github.com/camunda/camunda/issues/19833 for the Tasklist implementation, will be handled by me later on :)

**Comment by marcosgvieira** [comment]
@panagiotisgts @nathansandi what is left to merge this one?

**Comment by panagiotisgts** [comment]
@marcosgvieira it was a draft PR, the implementation should be complete now, waiting for a full review from @nathansandi

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml
Comment (nathansandi) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:883]: ❓ I see the default value is set here as 50, just to confirm, this is the case when there is no priority, or its null. this will assume 50?

**Review by nathansandi — APPROVED** [review]
Looks good to me, just a minor question.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml
Comment (panagiotisgts) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:883]: In the response, theoretically, you should never have null or unset priority since this is handled elsewhere. Mostly added this for documentation regarding the priority ranges.

**Pull request merged**
